instrumentation/opentelemetry/init.go

```diff
@@ -399,31 +393,25 @@ func initializeMetrics(cfg *config.AgentConfig) func() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	periodicReader := metric.NewPeriodicReader(metricsExporter)
 
 	resourceKvps := createResources(cfg.GetServiceName().GetValue(), cfg.ResourceAttributes)
 	resourceKvps = append(resourceKvps, identifier.ServiceInstanceKeyValue)
 	metricResources, err := resource.New(context.Background(), resource.WithAttributes(resourceKvps...))
 	if err != nil {
 		log.Fatal(err)
 	}
-	metricsPusher := controller.New(
-		processor.NewFactory(
-			simple.NewWithInexpensiveDistribution(),
-			metricsExporter,
-		),
-		controller.WithExporter(metricsExporter),
-		controller.WithResource(metricResources),
-	)
-	if err := metricsPusher.Start(context.Background()); err != nil {
-		log.Fatalf("starting metrics push controller: %v", err)
-	}
-
-	otelmetricglobal.SetMeterProvider(metricsPusher)
+	meterProvider := metric.NewMeterProvider(metric.WithReader(periodicReader), metric.WithResource(metricResources))
+	otelmetricglobal.SetMeterProvider(meterProvider)
 
 	return func() {
-		err := metricsPusher.Stop(context.Background())
+		err := periodicReader.Shutdown(context.Background())
+		if err != nil {
+			log.Printf("an error while calling metrics reader shutdown: %v", err)
+		}
+		err = meterProvider.Shutdown(context.Background())
```

Comment:
Should we invert the order of these 2 shutdowns since periodic reader was created first?

Reply:
I did it this way so that we would read all the latest metrics and push them out.  If it is flipped we will not have the last few right?

Reply:
Not sure I follow. If you shutdown the meter provider first, it means no one can create new meters dependent on the periodic reader we created.

Reply:
I'll switch it. I got thrown off in one of their tests. I read "reader" as just the collect.  It is both the collector and exporter.

Done.